lua-mosquitto.c

```diff
@@ -262,6 +262,37 @@ static int ctx_login_set(lua_State *L)
 	return mosq__pstatus(L, rc);
 }
 
+static int ctx_tls_opts_set(lua_State *L)
+{
+	ctx_t *ctx = ctx_check(L, 1);
+	int cert_reqs = luaL_optinteger(L, 2, 0);
+	const char *tls_version	 = luaL_optstring(L, 3, NULL);
+	const char *ciphers = luaL_optstring(L, 4, NULL);
+	
+	// the last param is a callback to a function that asks for a passphrase for a keyfile
+	// our keyfiles should NOT have a passphrase
+	int rc = mosquitto_tls_opts_set(ctx->mosq, cert_reqs, tls_version, ciphers);
+	return mosq__pstatus(L, rc);
+}
+
+static int ctx_mosquitto_version_set(lua_State *L)
+{
+	ctx_t *ctx = ctx_check(L, 1);
+	const char *mqtt_version = luaL_optstring(L, 2, NULL);
+	int protocol_version=MQTT_PROTOCOL_V31;
+	if(!strcmp(mqtt_version, "mqttv31"))
+	{
+		protocol_version = MQTT_PROTOCOL_V31;
+	}else if(!strcmp(mqtt_version, "mqttv311")){
+		protocol_version = MQTT_PROTOCOL_V311;
+	}
+	
+	// the last param is a callback to a function that asks for a passphrase for a keyfile
+	// our keyfiles should NOT have a passphrase
```

Comment:
Copy paste of the comment above?  What's right, what's wrong? Something needs correcting here.